contracts/truefi/TrueRatingAgencyV2.sol

```diff
@@ -379,15 +380,13 @@ contract TrueRatingAgencyV2 is ITrueRatingAgencyV2, Ownable {
      */
     modifier calculateTotalReward(address id) {
         if (loans[id].reward == 0) {
-            uint256 interest = ILoanToken(id).profit();
-
+            uint256 interest = ILoanToken2(id).profit();
             // calculate reward
             // prettier-ignore
             uint256 totalReward = toTRU(
-                interest
-                    .mul(distributor.remaining())
-                    .mul(rewardMultiplier)
-                    .div(distributor.amount())
+                interest.mul(distributor.remaining()).mul(rewardMultiplier).mul(10**18).div(distributor.amount()).div(
```

Comment:
Just double-checking for sanity here: is it okay that we can't call `decimals()` on LoanToken?

I think the original LoanToken imports ERC20 directly from OpenZeppelin, whereas LoanToken2 imports ERC20 from our UpgradeableERC20. I don't believe OZ's ERC20 has `decimals()` implemented. If we upgrade this TrueRatingAgencyV2 contract, wouldn't we be unable to interact with existing (or new legacy) LoanTokens?

Reply:
OZ's ERC20 actually does have decimals implemented and they it returns 18 as default.  Actually if You look at the tests, they run on both old LoanToken and the new LoanToken2 and they seem to pass just fine. Then again, Loans that are in review state right now are created by new LoanFactory2 so they are LoanTokens2 as well. I will not resolve this conversation if any further discussion is to appear. Anyway, I'm pretty confident that it works with both LoanToken and LoanToken2

Reply:
My bad, you're right! I just confirmed that OZ's ERC20 returns 18 by default. You also raise a good point that new loans should all be LoanToken2s because they come from LoanFactory2. Thanks for looking into this.